tests/ms2/processes/process-list.page.ts

```diff
@@ -165,20 +168,42 @@ export class ProcessListPage {
         await page.waitForURL('**/processes');
       }
 
+      /* Ensure nothing is selected (esc) */
+      await page.getByRole('main').click();
+      await page.getByRole('main').press('Escape');
+      /* Ensure that the search input is cleared */
+      const inputSearch = await page.locator('.ant-input-affix-wrapper');
+      await inputSearch.getByPlaceholder(/search/i).fill('');
+
       // make sure that the list is fully loaded otherwise clicking the select all checkbox will not work as expected
       await page.getByRole('columnheader', { name: 'Name' }).waitFor({ state: 'visible' });
 
-      // remove all processes
-      await page.getByLabel('Select all').check();
-      const modal = await openModal(this.page, () =>
-        page.getByRole('button', { name: 'delete' }).first().click(),
-      );
-      await closeModal(modal, () => modal.getByRole('button', { name: 'OK' }).click());
-
-      // Note: If used in a test, there should be a check for the empty list to
-      // avoid double navigations next.
-
-      this.processDefinitionIds = [];
+      while (this.processDefinitionIds.length > 0 || this.folderIds.length > 0) {
+        /* Get all currently visible processes */
+        const processRows = await page.locator('tr[data-row-key]').all();
+        const visibleIds = await asyncMap(processRows, async (el) =>
+          el.getAttribute('data-row-key'),
+        );
+
+        // remove all processes
+        await page.getByLabel('Select all').check();
+        const modal = await openModal(this.page, () =>
+          page.getByRole('button', { name: 'delete' }).first().click(),
+        );
+        await closeModal(modal, () => modal.getByRole('button', { name: 'OK' }).click());
+
+        const before = this.processDefinitionIds.length;
+        /* Remove entries from the process list */
+        this.processDefinitionIds = this.processDefinitionIds.filter(
+          (entry) => !visibleIds.includes(entry),
+        );
+        /* Remove entries from folder list */
+        this.folderIds = this.folderIds.filter((entry) => !visibleIds.includes(entry));
+
+        // Note: If used in a test, there should be a check for the empty list to
+        // avoid double navigations next.
+        // this.processDefinitionIds = [];
+      }
```

Comment:
Maybe an `expect` should be added, to check if there are still any processes / folders left.. but this would need an update of some test, as most do not update the `processListPage.processDefinitionIds` or `processListPage.folderIds`